tools/osv-scanner/main.go

```diff
@@ -252,5 +299,22 @@ func main() {
 		return
 	}
 
-	printResults(query, resp)
+	genericPrintResults(query, resp, func(query *osv.Query) bool {
```

Comment:
Can we push this complicated logic into `genericPrintResults` instead? There's a bit of complicated indirection and logic getting passed around with the current setup.